src/phoenix/trace/filter.py

```diff
@@ -1,21 +1,58 @@
 import ast
-from typing import Any, Iterator, Mapping, Tuple, cast
-
+import sys
+from difflib import SequenceMatcher
+from typing import (
+    Any,
+    Iterable,
+    Iterator,
+    Mapping,
+    Optional,
+    Protocol,
+    Sequence,
+    Tuple,
+    cast,
+)
+
+from typing_extensions import TypeGuard
+
+import phoenix.trace.v1 as pb
```

Comment:
this file used to not rely on the protos - it feels like that separation was a good one - it leaves the implementation to not be reliant on the protocol - notably the filtering code may not be in the same code base as the collection code.

Reply:
We could make a Python counterpart for it like we do for `span`. Making the switch itself is pretty trivial, it's literally just rewriting the piece of source code that's being parsed inside `_ast_evaluation_result_value`. On the other hand, this particular proto essentially behaves as nothing more than a container, but what we haven't done so far is to write up some helper functions to do the conversion (like we've done for span). We can open a ticket to do that later, because the only reason why this dependency exists is solely for type checking, which means to get rid of it we will need to change the type signatures in a few other places also (because it's a method of `evals` object).

Reply:
Not directly relying on the proto in the rest of the code would be my preference / advice as I think it keeps a clean separation of "business logic" from "transport" generated types. I think the idea of using the proto types throughout works great if there is a service that entirely communicates over protos but in our case I think it might be the case that we want to lift  / move the proto code. Understand it's a bit of duplication but I think it will give us flexibility down the line.

Reply:
I've opened a [ticket](https://github.com/Arize-ai/phoenix/issues/1865) to track this.